### PR TITLE
added: ProductListing page

### PR DESCRIPTION
**Pull request by tusharkandpal**
- added list of products in db
- ProductListing page consists of Sidebar & ProductCard components
- created ProductContext & productReducer
- `sidebar-toggle` class for responsiveness

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 
 function App() {
   return (
     <div className="App">
+      <Nav />
       <Routes>
         <Route path="/" element={<Home />}></Route>
+        <Route path="/product-listing" element={<ProductListing />}></Route>
         <Route path="/mockman" element={<MockMan />}></Route>
       </Routes>
```
Comment (dikshadhote) [inline src/App.js:15]: Can create separate file for route and then import it in App

**Review by dikshadhote — APPROVED** [review]
Overall neat and clean code. Really liked that you have taken care of all corner cases such as loading and error messages

**Review by neetukumari4858 — APPROVED** [review]
overall looks good and Code Structure is also good.

File: src/App.js (hunk 1 above)
Comment (tusharkandpal) [reply src/App.js:15]: i might do it after all paths are created. thanks for the suggestion.

**Pull request merged**